src/lib/src/grid3d/cell.cpp

```diff
@@ -99,4 +103,126 @@ cell_corners(const size_t i,
     return corners;
 }
 
+std::vector<double>
+get_corners_minmax(std::vector<double> &corners)
+{
+    double xmin = std::numeric_limits<double>::max();
+    double xmax = std::numeric_limits<double>::min();
+    double ymin = std::numeric_limits<double>::max();
+    double ymax = std::numeric_limits<double>::min();
+    double zmin = std::numeric_limits<double>::max();
+    double zmax = std::numeric_limits<double>::min();
+    for (auto i = 0; i < 24; i += 3) {
+        if (corners[i] < xmin) {
+            xmin = corners[i];
```

Comment:
Consider using `corners.at(i)` to ensure bounds checks happen here since the vector comes in unsized